cmd/snap-confine/mount-support.c

```diff
@@ -704,3 +704,45 @@ void sc_setup_user_mounts(struct sc_apparmor *apparmor, int snap_update_ns_fd,
 	sc_do_mount("none", "/", NULL, MS_REC | MS_SLAVE, NULL);
 	sc_call_snap_update_ns_as_user(snap_update_ns_fd, snap_name, apparmor);
 }
+
+void sc_ensure_snap_dir_shared_mounts(void)
+{
+	const char *dirs[] = { SNAP_MOUNT_DIR, "/var/snap", NULL };
+	for (int i = 0; dirs[i] != NULL; i++) {
+		const char *dir = dirs[i];
+		if (!is_mounted_with_shared_option(dir)) {
```

Comment:
Just thinking about this condition. I think this part is wrong a little. Let me explain.

Originally this code was fixing lack of sharing on /snap on distributions without systemd that ran snaps, aka ubuntu 14.04. Then the condition was correct since it was essentially checking if sharing is on or not.

With this change the purpose is two fold:

- to ensure that /snap is shared, because our mount namespace model requires it
- to make /var/snap and /snap mount points with for fine tuning the propagation of later

I think this responsibility is mixed up here. On a systemd-less distribution and on a systemd-based distribution alike, we want them to be mount points.

Reply:
After some IRC chatter I think this is still problematic. If this is not the 14.04 special case then the condition is wrong. The correct condition should be: absence of a mount point at this spot.

Reply:
Ok, adding a check for a mountpoint.

Reply:
Per later discussion, we're leaving the shared mountpoint check here to keep things simple.

As an extension, #7570 introduces a mount unit to do it automatically at boot for $SNAP_MOUNT_DIR.